chipy_org/apps/job_board/models.py

```diff
@@ -130,3 +122,47 @@ def days_elapsed(self):
             return days_elapsed_from_posting
         else:
             return None
+
+    @property
+    def expiration_date(self):
+
+        # expiration_date shows up as a field in the admin panel.
+        # If post is approved and the approval_date is set,
+        # compute the expiration_date.
+        if self.status == "AP" and self.approval_date:
+            expiration_date = self.approval_date + datetime.timedelta(days=self.days_to_expire)
+            return expiration_date
+        else:
+            return None
+
+    @classmethod
+    def approved_posts(cls):
+
+        # I've split these into two queries in anticipating that there might be
+        # different ordering or filtering based on sponsored vs non-sponsored
+        # job posts.
+
+        now = datetime.datetime.now()
+
+        # For the 3rd Q()in sponsored_job_post and other_job_posts, I tried to do
+        # datetime.timedelta(days=F('days_to_expire')) but you
+        # can't put a F() inside of timedelta.
+        # So instead I did datetime.timedelta(days=1)*F('days_to_expire')).
+
+        sponsored_job_posts = JobPost.objects.filter(
+            Q(status="AP")
+            & Q(is_sponsor=True)
+            & Q(approval_date__gte=now - (datetime.timedelta(days=1) * F("days_to_expire")))
+        ).order_by("-approval_date")
+
+        other_job_posts = JobPost.objects.filter(
+            Q(status="AP")
+            & Q(is_sponsor=False)
+            & Q(approval_date__gte=now - (datetime.timedelta(days=1) * F("days_to_expire")))
+        ).order_by("-approval_date")
+
+        # I put the two groups of job posts back together so they can processed
+        # by the same loop in the template.
+        job_posts = list(chain(sponsored_job_posts, other_job_posts))
```

Comment:
Do you need to break up the queries? Could you use group_by to group the sponsored and other job posts.  

Another way may be three method.  The first method would be `approved` to get all the approved posts.   Then you use  two more methods which are `approved_and_sponsored` and `approved_and_not_sponsored' which rely on the code from `approved`.   Then you can easily make the calls below in your code.    

1. JobPost.approved.all()
2. JobPost.approved_and_sponsored.all()
3. JobPost.approved_and_not_sponsored.all()

**Side Note:**
@raymondberg  and I both asked you to use a classmethod for this.  However, I recently found out about  [Django's Manager Class](https://docs.djangoproject.com/en/3.1/topics/db/managers/).  I won't ask you to refactor using the Django manager.  I do want to point out what seems to be the "Django way" of doing this task.